### PR TITLE
Hw1

### PR DESCRIPTION
**Pull request by Rebanaito**
Implemented Tasks 0-8, as well as tests for 1-8 and utility functions gathered in a separate class. Tests cover most common scenarios as well as some edge cases and invalid inputs. All code was formatted to pass the style check.

### CODE REVIEW

File: src/main/java/edu/hw1/Task_6.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
+package edu.hw1;
+
+import java.util.Arrays;
+
+@SuppressWarnings("TypeName")
+public final class Task_6 {
+    private Task_6() {
+    }
+
+    private static boolean sameDigits(int n) {
```
Comment (sirvaulterscoff) [inline src/main/java/edu/hw1/Task_6.java:10]: логичнее располагать членов класса в следующем порядке:
- конструкторы
- публичные методы
- protected
- private

При чтении кода обычно в первую очередь тебя интересуют именно публичные методы - внешний контракт класса. А приватные члены класса обычно выполняют вспомогательную функцию

File: src/main/java/edu/hw1/Utils.java [hunk 2]
```diff
@@ -0,0 +1,85 @@
+package edu.hw1;
+
+public final class Utils {
+    private Utils() {
+    }
+
+    @SuppressWarnings("MagicNumber")
+    private static void itoaInternal(StringBuilder builder, int num) {
```
Comment (sirvaulterscoff) [inline src/main/java/edu/hw1/Utils.java:8]: В Java есть готовые методы преобразования числа в строки, и наоборот
Integer.toString()
String.valueOf(num)
Для более сложных кейсов есть NumberFormat

File: src/main/java/edu/hw1/Utils.java [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+        if (array == null) {
+            return 0;
+        }
+        if (array.length == 0) {
+            return 0;
+        }
+        return atoiInternal(array);
+    }
+
+    public static boolean isNum(String str) {
```
Comment (sirvaulterscoff) [inline src/main/java/edu/hw1/Utils.java:69]: Integer.valueOf + try/catch(NumberFormatException)
